webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/properties/PropertiesRepository.scala

```diff
@@ -52,12 +54,12 @@ import zio.*
 
 trait PropertiesRepository {
 
-  def getAllNodeProps()(implicit qc: QueryContext): IOResult[Map[NodeId, Chunk[NodePropertyHierarchy]]]
+  def getAllNodeProps()(implicit qc: QueryContext): IOResult[Map[NodeId, ResolvedNodePropertyHierarchy]]
 
   /*
    * Get all properties for node with given ID
    */
-  def getNodeProps(nodeId: NodeId)(implicit qc: QueryContext): IOResult[Option[Chunk[NodePropertyHierarchy]]]
+  def getNodeProps(nodeId: NodeId)(implicit qc: QueryContext): IOResult[ResolvedNodePropertyHierarchy]
```

Comment:
It never happens appart in extrem case that people actually deconstruct errors to recover from them, because they are hidden by the api and so, as a dev, you chose to hide them from the easy actionable path for the recipient.
And really, at that point it's not an error to not have the node, it's just not here, and the caller will decide if it's a error for him

Reply:
And as a rule, error passed in the error chan in io result are defects, never part of the expected outputs of domain model.
Only the last layer can decide if it's an error.